File: 백준/Silver/10773. 제로/제로.py
from collections import deque
K = int(input())
deque = deque()
answer = 0
for i in range(K):
    now = int(input())
    if now == 0:
        answer -= deque.pop()
    else:
        deque.append(now)
        answer += now
print(answer)

File: 백준/Silver/10773. 제로/제로.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

const K = input[0];
const stack = [];

let answer = 0;
for(let i = 1; i <= K; i++){
    let now = Number(input[i]);
    if(now == 0){
        answer -= stack.pop();
    }else{
        stack.push(now);
        answer+= now;
    }
}
console.log(answer);